jawa/attributes/code.py

```diff
@@ -168,6 +170,17 @@ def disassemble(self, *, transforms=None):
         elif transforms is None:
             transforms = []
 
+        for i, transform in enumerate(transforms):
+            sig = inspect.signature(transform, follow_wrapped=True)
+            kws = {}
+            if 'cf' in sig.parameters:
+                kws['cf'] = self.cf
+
+            if 'attribute' in sig.parameters:
+                kws['attribute'] = self
+
+            transforms[i] = functools.partial(transform, **kws)
```

Comment:
Because of the earlier

```python
        if transforms is None and self.cf.classloader:
            transforms = self.cf.classloader.bytecode_transforms
```

this can modify the classloader's `bytecode_transforms`.  While this won't cause any _obvious_ side affects at first (it'll just keep overwriting the default parameter), this does cause issues if you have a call to disassemble, and then within that call end up calling it a second time; when the inner call ends the outer call will start using the classfile from the inner call.

One possible fix is to use `transforms = list(self.cf.classloader.bytecode_transforms)` but this is fairly hacky (this is the approach I'm using temporarily and it does work).  Another would be to use a list comprehension, which would solve this issue entirely (modifying the passed in `transforms` value is confusing for other reasons -- e.g. if the user were passing in a single shared list) but might be a bit ugly.